packages/react/src/useFindBy.ts

```diff
@@ -79,12 +79,12 @@ export const useFindBy = <
 
   let error = ErrorWrapper.forMaybeCombinedError(result.error);
   if (!error) {
-    const nonNullableError = getNonNullableError(result, dataPath);
-
     if (records.length > 1) {
       error = ErrorWrapper.forClientSideError(getNonUniqueDataError(finder.modelApiIdentifier, finder.findByVariableName, value));
-    } else if (nonNullableError) {
-      error = ErrorWrapper.forClientSideError(nonNullableError);
+    } else if (result.data && !records[0]) {
+      error = ErrorWrapper.forClientSideError(
+        new GadgetNotFoundError(`${finder.modelApiIdentifier} record with ${finder.findByVariableName}=${value} not found`)
+      );
```

Comment:
We documented `useFindBy` as throwing if the record isn't found for the field value, but it didn't actually do that. I made it do it... I think we want that?

Reply:
Would we also want something like a `useMaybeFindBy`? Just wondering if throwing forces people into something that doesn't fit the way they're building their react apps (e.g., Suspense / `ErrorBoundary` / preferred `error` value)